projects/isy-angular-widgets/src/lib/validation/validation.spec.ts

```diff
@@ -26,110 +26,114 @@ describe('Unit Test: Validation', () => {
+    ];
+
+    invalidInputs.forEach(({input, description}) => {
+      it(`should return ${errorKey} for ${description}`, () => {
+        control.setValue(input);
+        const errors = Validation.validUnspecifiedDate(control);
+        expect(errors).toEqual({[errorKey]: true});
+      });
     });
+  });
 
-    it('should return UNSPECIFIEDDATE if date is unspecified and the month is invalid in german date format xx.MM.YYYY', () => {
-      const control: AbstractControl = new FormControl('xx.13.2023');
-      const errors = Validation.validUnspecifiedDate(control);
-      errorHaveToBeDefined(errors, errorKey);
-    });
+  describe('validUnspecifiedISODate when input is valid', () => {
+    let control: AbstractControl;
 
-    it('should return UNSPECIFIEDDATE if date is in german date format DD.xx.2022', () => {
-      const control: AbstractControl = new FormControl('01.xx.2023');
-      const errors = Validation.validUnspecifiedDate(control);
-      errorHaveToBeDefined(errors, errorKey);
+    beforeEach(() => {
+      control = new FormControl('');
     });
 
-    it('should return UNSPECIFIEDDATE if date is in german date format xx.MM.xxxx', () => {
-      const control: AbstractControl = new FormControl('xx.01.xxxx');
-      const errors = Validation.validUnspecifiedDate(control);
-      errorHaveToBeDefined(errors, errorKey);
-    });
+    const validInputs = [
+      '',
+      '2023-11-11',
+      '2023-11-00',
+      '2023-00-00',
+      '0000-00-00',
+      '2023-11-xx',
+      '2023-xx-xx',
+      'xxxx-xx-xx'
+    ];
 
-    it('should return UNSPECIFIEDDATE if date is in german date format DD.xx.xxxx', () => {
-      const control: AbstractControl = new FormControl('01.xx.xxxx');
-      const errors = Validation.validUnspecifiedDate(control);
-      errorHaveToBeDefined(errors, errorKey);
+    validInputs.forEach((input) => {
+      it(`should return null for '${input}'`, () => {
+        control.setValue(input);
+        const errors = Validation.validUnspecifiedISODate(control);
+        expect(errors).toBeNull();
+      });
     });
+  });
 
-    it('should return UNSPECIFIEDDATE if date is in german date format DD.MM.xxxx', () => {
-      const control: AbstractControl = new FormControl('01.01.xxxx');
-      const errors = Validation.validUnspecifiedDate(control);
-      errorHaveToBeDefined(errors, errorKey);
+  describe('validUnspecifiedISODate when input is invalid', () => {
+    let control: AbstractControl;
+    const errorKey = 'UNSPECIFIEDISODATE';
+
+    beforeEach(() => {
+      control = new FormControl('');
+    });
+
+    const testCases = [
+      {input: '2023-11-50', desc: 'invalid day'},
+      {input: '2023-30-01', desc: 'invalid month'},
+      {input: '2023-13-00', desc: 'invalid month with unspecified day'},
+      {input: '2023-13-xx', desc: 'invalid month in unspecified date'},
+      {input: '2023-xx-01', desc: 'valid day in unspecified month and year'},
```

Comment:
missing testCase:
{input: '2023-00-01', desc: 'valid day and year with unspecified month'},

Reply:
added